.github/actions/update-config/action.yaml

```
@@ -48,58 +48,56 @@ runs:
       shell: bash
+      env:
+        GH_TOKEN: ${{ inputs.cicd_token }}
       id: create_pr
       run: |
         cd config-repo
-        PR_TITLE="chore: update configs for ${{ inputs.networks }} from ${{ inputs.source_repo }}"
+        PR_TITLE="chore: update configs for ${{ inputs.network }} from ${{ inputs.source_repo }}"
         PR_BODY="Auto-generated PR with updated configuration files"
         PR_HEAD="${{ steps.branch-name.outputs.branch }}"
         PR_BASE="main"
-        # Capture both response and HTTP status code
-        RESPONSE=$(curl -s -w "%{http_code}" -o response.json -X POST \
-          -H "Authorization: token ${{ inputs.cicd_token }}" \
-          -H "Accept: application/vnd.github.v3+json" \
-          https://api.github.com/repos/${{ inputs.config_repo }}/pulls \
-          -d "{\"title\":\"$PR_TITLE\",\"body\":\"$PR_BODY\",\"head\":\"$PR_HEAD\",\"base\":\"$PR_BASE\"}")
-        HTTP_STATUS=$(echo -n "$RESPONSE" | tail -c 3)
-        cat response.json
-        PR_NUMBER=$(jq -r .number response.json)
-        if [[ "$HTTP_STATUS" != "201" ]] || [[ "$PR_NUMBER" == "null" || -z "$PR_NUMBER" ]]; then
-          echo "Failed to create PR. HTTP status: $HTTP_STATUS, PR number: $PR_NUMBER"
-          cat response.json
-          exit 1
+
+        echo "Creating PR: $PR_TITLE"
+        CREATE_PR_URL=$(gh pr create \
+          --repo "${{ inputs.config_repo }}" \
+          --title "$PR_TITLE" \
+          --body "$PR_BODY" \
+          --head "$PR_HEAD" \
+          --base "$PR_BASE")
+        CREATE_PR_EXIT_CODE=$?
+
+        if [ "$CREATE_PR_EXIT_CODE" -ne 0 ]; then
+            echo "Failed to create PR."
+            exit 1
         fi
+        PR_NUMBER=$(gh pr view "$CREATE_PR_URL" --repo "${{ inputs.config_repo }}" --json number -q .number)
         echo "PR_NUMBER=$PR_NUMBER" >> "$GITHUB_ENV"
 
     - name: Merge Pull Request
       shell: bash
+      env:
+        GH_TOKEN: ${{ inputs.cicd_token }}
       run: |
         cd config-repo
         PR_NUMBER=${{ env.PR_NUMBER }}
-        PR_TITLE="chore: update configs for ${{ inputs.networks }} from ${{ inputs.source_repo }}"
-        # Attempt to merge the PR and capture both response and HTTP status code
-        RESPONSE=$(curl -s -w "%{http_code}" -o merge_response.json -X PUT \
-          -H "Authorization: token ${{ inputs.cicd_token }}" \
-          -H "Accept: application/vnd.github.v3+json" \
-          https://api.github.com/repos/${{ inputs.config_repo }}/pulls/$PR_NUMBER/merge \
-          -d "{\"commit_title\":\"Merge PR #$PR_NUMBER: $PR_TITLE\",\"merge_method\":\"squash\"}")
-        HTTP_STATUS=$(echo -n "$RESPONSE" | tail -c 3)
-        cat merge_response.json
-        if [[ "$HTTP_STATUS" != "200" ]]; then
-          echo "Failed to merge PR. HTTP status: $HTTP_STATUS"
-          cat merge_response.json
+        PR_TITLE="chore: update configs for ${{ inputs.network }} from ${{ inputs.source_repo }}"
+        gh pr merge "$PR_NUMBER" \
```

```suggestion

        gh pr merge "$PR_NUMBER" \
```
use newlines between variable declarations and cmds for readability